### PR TITLE
Feat/dq update status tags in main view  issue lists

### PR DESCRIPTION
**Pull request by mariaozamiz**
### :pushpin: References

-   **Issue:** Closes [DQ: Update Status tags in Main view + Issue lists](https://app.clickup.com/t/86944v8m4)
- CU-86944v8m4

### :memo: Implementation

Create Tag component for dashboard table.
Create ButtonTag for issues table.

### :video_camera: Screenshots/Screen capture

Captura de pantalla 2024-03-25 a las 11 58 09

https://github.com/EyeSeeTea/data-quality-dev/assets/55712984/f42b677b-ba02-4614-9ce6-7503222a4fd8

### CODE REVIEW

**Comment by ifoche** [comment]
Task linked: [CU-86944v8m4 DQ: Update Status tags in Main view + Issue lists](https://app.clickup.com/t/86944v8m4)

**Review by Ramon-Jimenez — APPROVED** [review]
Looking great. Waiting for Eduardo's feedback from coding perspective before merging.

File: src/webapp/components/tag/ButtonTag.tsx
Comment (eperedo) [inline src/webapp/components/tag/ButtonTag.tsx:20]: This condition should be handled in the parent component.

```tsx
// MyCustomComponent.tsx
<ButtonTag text={text ? text : "No Action"} onClick={onOpenMenu} />
```

Now we don't have a hardcoded value inside our component, making it more easy to reuse.

**Review by eperedo — CHANGES_REQUESTED** [review]
Thanks for the code @mariaozamiz. Some comments to keep components more reusable. Let me know if you have any questions

**Comment by mariaozamiz** [comment]
@eperedo what do you think about this piece holding the "No action" text instead of the ButtonTag? Maybe we can set at this selector a default no-action state to show in a different way?
`  <SelectorInline
                    items={metadata.optionSets.nhwaAction.options.map(option => {
                        return { id: option.code, label: option.name };
                    })}
                    value={issue.action?.name || "No action"}
                    onChange={onSave}
                />`

**Comment by eperedo** [comment]
> @eperedo what do you think about this piece holding the "No action" text instead of the ButtonTag? Maybe we can set at this selector a default no-action state to show in a different way? ` <SelectorInline items={metadata.optionSets.nhwaAction.options.map(option => { return { id: option.code, label: option.name }; })} value={issue.action?.name || "No action"} onChange={onSave} />`

We cannot use by default the "No action" text because it's a value inside the optionSet. And the requirement was to keep this field empty for new issues.

image

File: src/webapp/utils/analysis.tsx
Comment (eperedo) [inline src/webapp/utils/analysis.tsx:9]: I think we should use another function `mapIssueActionStatusToColor` since the analysis can have different values. It could be confusing if we mix all the possible values.

File: src/webapp/components/tag/Tag.tsx
Comment (eperedo) [inline src/webapp/components/tag/Tag.tsx:3]: No need to change, but normally import the whole React object and then we use `React.memo` in the code. The advantage is to have a shorter import.

**Review by eperedo — CHANGES_REQUESTED** [review]

**Comment by mariaozamiz** [comment]
Couldn't "No action" be a default value for Actions? What I'm trying to address is this empty button. Setting the "No action" text as an alternative at the ButtonTag brings up other issues, because the component is being used both at Action and Status columns. 

Captura de pantalla 2024-04-01 a las 9 33 04

File: src/webapp/utils/analysis.tsx
Comment (mariaozamiz) [reply src/webapp/utils/analysis.tsx:9]: done!

**Comment by eperedo** [comment]
> Couldn't "No action" be a default value for Actions? What I'm trying to address is this empty button. Setting the "No action" text as an alternative at the ButtonTag brings up other issues, because the component is being used both at Action and Status columns.
> 
> Captura de pantalla 2024-04-01 a las 9 49 34 Captura de pantalla 2024-04-01 a las 9 33 04

As I remember correctly - @Ramon-Jimenez  please confirm this - for the client "No actions" means that they validated the issue and decided to... well take no actions 😄. Not sure if we can change this behaviour. Adding the "No action" value for new issues? @Ramon-Jimenez

**Review by Ramon-Jimenez — CHANGES_REQUESTED** [review]
We can go ahead with the approach suggested by  @mariaozamiz  and make No Action the default one. It is true that it was supposed to be used by the client when the item was reviewed and No Action needed to take place, but it also makes sense that, by default, all items have No Action assigned (same way all items have 'Not treated' status also by default).

Thanks @mariaozamiz  @eperedo

**Comment by mariaozamiz** [comment]
@eperedo can you have another look please? 🙏 New changes have been made according to the group chat today

**Review by eperedo — CHANGES_REQUESTED** [review]
Looks good @mariaozamiz. Just a minor change to set the action for new issues using the default action.

```ts
// UCIssue.ts
    buildDefaultIssue(values: DefaultIssue, sectionId: Id): QualityAnalysisIssue {
        return new QualityAnalysisIssue({
            // pass the code "0" for the action property
            // this is the code for the "No action" value
            action: IssueAction.create({
                code: "0",
                id: "",
                name: "",
            }),
            ...
        });
    }
```

That way every time we create a new issue it's going to have the "No Action" value by default and we don't need to pass the string in the `EditIssueValue` component

**Review by eperedo — APPROVED** [review]
Thanks @mariaozamiz!

**Review by Ramon-Jimenez — APPROVED** [review]
Thx @mariaozamiz

**Pull request merged**
